example/matrixMulWithMdspan/src/matrixMulMdSpan.cpp

```diff
@@ -0,0 +1,178 @@
+/* Copyright 2023 Mehmet Yusufoglu, Bernhard Manfred Gruber, René Widera, Andrea Bocci, Simeon Ehrig
+ * SPDX-License-Identifier: ISC
+ */
+
+#include <alpaka/alpaka.hpp>
+// Needed for running example for all backends available; one by one
+#include <alpaka/example/ExecuteForEachAccTag.hpp>
+
+#include <experimental/mdspan>
+#include <iostream>
+
+// Index type
+using Idx = std::size_t;
+// Set data type
+using DataType = float;
+
+/**
+ * @brief Kernel for performing multiplication of two 2D matrices. Each element is computed by a different thread.
+ * MdSpan data structure is used to pass the data to and from the kernel.
+ */
+struct MatrixMulKernel
+{
+    //! \tparam TAcc Accelerator type
+    //! \tparam MdSpan The type of the multidimensional span (mdspan)
+    //! \param acc Accelerator
+    //! \param A First input matrix
+    //! \param B Second input matrix
+    //! \param C Output matrix where the result of A * B will be stored
+    //! \param K The shared dimension between A and B
+    template<typename TAcc, typename MdSpan>
+    ALPAKA_FN_ACC void operator()(TAcc const& acc, MdSpan A, MdSpan B, MdSpan C, Idx K) const
```

Comment:
shouldn't `K` be derived from `A` and `B`, with a check that the dimensions match ?

Reply:
Good catch. Thanks done.